redis-snapshot-replicator/functions/shipper.py

```diff
@@ -0,0 +1,32 @@
+import boto3
+import botocore
+import datetime
+import re
+import os
+
+iam = boto3.client('iam')  
+instances = os.environ['DB_INSTANCES']  
+target_bucket = os.environ['TARGET_BUCKET'] 
+
+print('Loading function')
+
+def byTimestamp(snap):  
+    if 'SnapshotCreateTime' in snap:
+        return datetime.datetime.isoformat(snap['SnapshotCreateTime'])
+    else:
+        return datetime.datetime.isoformat(datetime.datetime.now())
+
+def lambda_handler(event, context): 
+
+    source = boto3.client('elasticache')
+
+    for instance in instances.split(','):
+        source_snaps = source.describe_snapshots(CacheClusterId=instance)['Snapshots']
+        source_snap = sorted(source_snaps, key=byTimestamp, reverse=True)[0]['SnapshotName']
+        
+        try:
+            response = source.copy_snapshot(SourceSnapshotName=source_snap,TargetSnapshotName=source_snap,TargetBucket=target_bucket)
```

Comment:
Why do you need a lambda function to copy snapshots from one bucket to the other if you've already enabled bucket replication? 🤔

Reply:
The copy snapshot command copies the snapshot from the "redis backup bucket" managed by AWS to a self created bucket. This has nothing to do with the replication

Reply:
ok, got it